### ISSUE TITLE
core: extend deriation path for dex to address suspended accounts

### ISSUE THREAD

**Issue by chappjc** [decred/dcrdex#1204]
In https://github.com/decred/dcrdex/pull/1201, I was dealing with the issue of suspended accounts and registration with a legacy key fallback.  I think this should be changed, as has been suggested previously, to derive new keys that branch at the DEX host's path.

Demonstrated on that PR, but the behavior on master is the same:

```
[INF] CORE: HD account key for 127.0.0.1:17273 was reported as suspended. Register to make a new one.
"POST http://127.0.0.3:5758/api/discoveracct HTTP/1.1" from 127.0.0.1:40232 - 200 968B in 1.247533373s
```

takes to Confirm screen with the markets table, then click confirm for it to call Register, which discovers the HD key won't work and makes a legacy account...

```
[INF] CORE: HD account key for 127.0.0.1:17273 was reported as suspended.
[INF] CORE: Generating a legacy-style key to register with 127.0.0.1:17273.
```

While this is the current behavior on master, it does make me a little concerned that the user might not realize it went the "legacy" account path here, meaning the account cannot be recovered by the seed, only export/import.  I think this just means we probably do want to pursue multiple child keys at the derivation path for a given DEX host.

_Originally posted by @chappjc in https://github.com/decred/dcrdex/pull/1201#r705735386_

**Comment by itswisdomagain** [comment]
> we probably do want to pursue multiple child keys at the derivation path for a given DEX host

Agree.

One thought is to use child keys of the original DEX account key, starting at 1 until a non-suspended account is found, i.e.
- where the original/default DEX account key continue to use the path **`hdPurpose' / dexPK[0]' / dexPK[1:5]' / dexPK[5:9]' / dexPK[9:14]' / dexPK[14:19]' / dexPK[19:24]' / dexPK[24:29]' / dexPK[29:33]'`**
- other accounts for the same DEX use the extended path **`hdPurpose' / dexPK[0]' / dexPK[1:5]' / dexPK[5:9]' / dexPK[9:14]' / dexPK[14:19]' / dexPK[19:24]' / dexPK[24:29]' / dexPK[29:33]' / i'`** (where 1 >= i <= 2^31)
to prevent increasing the possibility of collision.

**Comment by JoeGruffins** [comment]
I agree. I believe I commented on the initial pr, just add one to the child index until you find an account that is not suspended. It should end up being the same account every restore as long as it too isn't suspended. Maybe oversimplifying.

**Comment by chappjc** [comment]
Did you want to tackle this @itswisdomagain ?

**Comment by itswisdomagain** [comment]
I believe I can whip up something pretty quickly. Yes please.

**Comment by chappjc** [comment]
> I agree. I believe I commented on the initial pr, just add one to the child index until you find an account that is not suspended. It should end up being the same account every restore as long as it too isn't suspended. Maybe oversimplifying.

Yeah, seems that "simple" to me @JoeGruffins
